### PR TITLE
UI test cases for IOT server

### PR DESCRIPTION
**Pull request by menakaj**
Test scenarios:
Base UI test cases:
- Login/ Logout
- New User registration
- Adding a user by admin
- User deletion
- Creating device groups 

Sample Testing
- Sample installation and verification.
- Sample enrollment and verification.
- Graphs verification.
- Data publishing verification.

### CODE REVIEW

**Comment by ruwany** [comment]
Merging tests by Menaka

File: modules/integration/tests-common/web-ui-pages/src/main/java/org/wso2/iot/integration/ui/pages/devices/DevicesPage.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+     * This method performs the navigation to the Device view of the given device.
+     * Here the navigation happens to the Connected cup device.
+     * @param deviceName : Name of the device.
+     * @return : The corresponding device view page. Null if not visible.
+     */
+    public ConnectedCupDeviceViewPage viewDevice(String deviceName) throws IOException {
+        WebElement deviceTable = driver.findElement(By.xpath(uiElementMapper.getElement("iot.devices.table.xpath")));
+        List<WebElement> data = deviceTable.findElements(By.cssSelector("a"));
+        for (WebElement e : data) {
+            String s = getLink(e, "/device/connectedcup?id=");
```
Comment (inoshperera) [inline modules/integration/tests-common/web-ui-pages/src/main/java/org/wso2/iot/integration/ui/pages/devices/DevicesPage.java:85]: Can we use a meaningful variable name here?

File: modules/integration/tests-common/web-ui-pages/src/main/java/org/wso2/iot/integration/ui/pages/devices/EnrollDevicePage.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        boolean check = UIUtils.isElementPresent(log, driver, By.xpath(
+                uiElementMapper.getElement("iot.sample.connectedcup.xpath")));
+        if (check){
+            WebElement deviceDiv = driver.findElement(By.xpath(
+                    uiElementMapper.getElement("iot.sample.connectedcup.xpath")));
+            WebElement tryBtn = deviceDiv.findElement(By.tagName("button"));
+            tryBtn.click();
+            return new ConnectedCupDeviceTypeViewPage(driver);
+        } else {
+            log.error("Element not found...........................");
```
Comment (inoshperera) [inline modules/integration/tests-common/web-ui-pages/src/main/java/org/wso2/iot/integration/ui/pages/devices/EnrollDevicePage.java:84]: Can we make the error log standerd

File: modules/integration/tests-common/web-ui-pages/src/main/java/org/wso2/iot/integration/ui/pages/devices/DevicesPage.java (hunk 1 above)
Comment (menakaj) [reply modules/integration/tests-common/web-ui-pages/src/main/java/org/wso2/iot/integration/ui/pages/devices/DevicesPage.java:85]: This is fixed in the commit https://github.com/wso2/product-iots/pull/1454/commits/399136daae1b5183134e5b3da6c5fef9fa633f47

File: modules/integration/tests-common/web-ui-pages/src/main/java/org/wso2/iot/integration/ui/pages/devices/EnrollDevicePage.java (hunk 2 above)
Comment (menakaj) [reply modules/integration/tests-common/web-ui-pages/src/main/java/org/wso2/iot/integration/ui/pages/devices/EnrollDevicePage.java:84]: This is fixed in the commit https://github.com/wso2/product-iots/pull/1454/commits/399136daae1b5183134e5b3da6c5fef9fa633f47

**Pull request merged**